### PR TITLE
Npc Buffer cleanup. Move spells to conf

### PR DESCRIPTION
**Pull request by Sdyess**
The player should cast the buffs on himself, there are weird problems when the creature attempts to do it. This change removes the repetitiveness and makes it easier to add and remove spells through the conf file.

Tested & working for me

### CODE REVIEW

File: src/npc_buffer.cpp [hunk 1]
```diff
@@ -87,28 +87,26 @@ class buff_npc : public CreatureScript
...
         // Apply Buffs
-        creature->CastSpell(player, Buff1, true);
-        creature->CastSpell(player, Buff2, true);
-        creature->CastSpell(player, Buff3, true);
-        creature->CastSpell(player, Buff4, true);
-        creature->CastSpell(player, Buff5, true);
-        creature->CastSpell(player, Buff6, true);
+
+        for (std::vector<uint32>::const_iterator itr = vecBuffs.begin(); itr != vecBuffs.end(); itr++)
+            player->CastSpell(player, *itr, true);
```
Comment (talamortis) [inline src/npc_buffer.cpp]: Love it! players can add more spells than just the 6th.

File: src/npc_buffer.cpp [hunk 2]
```diff
@@ -119,6 +117,7 @@ class buff_npc : public CreatureScript
     }
 };
 
+
```
Comment (talamortis) [inline src/npc_buffer.cpp]: can you remove this extra line?

File: src/npc_buffer.cpp [hunk 3]
```diff
@@ -142,9 +141,10 @@ class npc_buffer : public WorldScript
...
```
Comment (talamortis) [inline src/npc_buffer.cpp]: and this

File: src/npc_buffer.cpp [hunk 4]
```diff
@@ -87,28 +87,26 @@ class buff_npc : public CreatureScript
...
+        std::stringstream ss(sConfigMgr->GetStringDefault("Buff.Spells", ""));
+
+        for (std::string buff; std::getline(ss, buff, ';');)
+        {
+            vecBuffs.push_back(stoul(buff));
+        }
 
         // Remove Ressurection Sickness?
-        if (sConfigMgr->GetBoolDefault("Buff.CureRes", true))
+        if (sConfigMgr->GetBoolDefault("Buff.CureRes", 0))
```
Comment (talamortis) [inline src/npc_buffer.cpp]: please revert this change, if bool = false the give the ress sick option?

File: src/npc_buffer.cpp [hunk 5]
```diff
@@ -59,7 +59,7 @@ Creates a one-click Buff NPC with emotes.
 
 #include "Config.h"
 #include "ScriptPCH.h"
-#include "Configuration/Config.h"
```
Comment (talamortis) [inline src/npc_buffer.cpp]: this needs to stay for the sScriptMgr for config

File: src/npc_buffer.cpp [hunk 6]
```diff
@@ -59,7 +59,7 @@ Creates a one-click Buff NPC with emotes.
...
+
```
Comment (talamortis) [inline src/npc_buffer.cpp]: remove this extra line

**Comment by Sdyessdev** [comment]
Sorry, I was sloppy when i stashed my changes to add yours and the auto merge didn’t work :P Will add a commit in a couple hours for it
